src/adapter/etl-adapter-excel/src/Flow/ETL/Adapter/Excel/ExcelExtractor.php

```diff
@@ -61,10 +61,12 @@ public function extract(FlowContext $context) : \Generator
 
         foreach ($context->streams()->list($this->path, $this->filter()) as $stream) {
             foreach ($this->extractRows($stream, $headers, $offset) as $row) {
-                $signal = yield array_to_rows($row, $context->entryFactory());
+                $signal = yield rows(array_to_row($row, $context->entryFactory(), $stream->path()->partitions()));
```

Comment:
it seems that there is a missing test here, no?

Reply:
I will try to generate a file that can address such an issue, it should be similar to the one used now when the last row has an empty cell.

Reply:
I was more refering to the fact that nothing is checking if partition is available at yielded Rows, which is pretty easy to reproduce in tests by testing reading from pratitioned dataset